common/scala/src/main/scala/whisk/core/database/CouchDbRestStore.scala

```diff
@@ -61,6 +67,13 @@ class CouchDbRestStore[DocumentAbstraction <: DocumentSerializer](
   private val client: CouchDbRestClient =
     new CouchDbRestClient(dbProtocol, dbHost, dbPort.toInt, dbUsername, dbPassword, dbName)
 
+  // This the the amount of allowed parallel requests for each entity, before batching starts. If there are already maxOpenDbRequests
+  // and more documents need to be stored, then all arriving documents will be put into batches (if enabled) to avoid a long queue.
+  private val maxOpenDbRequests = system.settings.config.getInt("akka.http.host-connection-pool.max-open-requests") / 2
+
+  private val batcher: Batcher[JsObject, Either[ArtifactStoreException, DocInfo]] =
+    new Batcher(500, maxOpenDbRequests)(put(_)(TransactionId.unknown))
```

Comment:
Seems like the batch size should be configurable instead of hardcoded.

Reply:
Why do you think it should be configurable? Do you have a scenario in mind, where you want to configure this size?
Btw: It is only the maximum allowed batch size.

When I was chatting with Cloudant developpers a while ago, we were talking about such bulk requests. The scenario was a bit different, but very similar (write documents as fast as possible into a database with _bulk). This person suggested a batch size of 500.

Reply:
Was thinking the value might change per deployment. Should at least add a comment stating why it's set to 500 here.